cmd/envtool/tests.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+)
+
+// shardIntegrationTests shards integration test names from the specified path.
+func shardIntegrationTests(w io.Writer, index, total uint) error {
+	var output strings.Builder
+
+	testNames, err := shardTests(index, total, "integration")
+	if err != nil {
+		return err
+	}
+	shardedTestNames := testNames[index:total]
+
+	output.WriteString("^(")
+	output.WriteString(strings.Join(shardedTestNames, "|"))
+	output.WriteString(")$")
+
+	w.Write([]byte(output.String()))
+
+	return nil
+}
+
+// shardTests shards gotten test names from the specified path.
+func shardTests(index, total uint, path string) ([]string, error) {
+	testNames, err := getAllTestNames(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return testNames[index:total], nil
```

Comment:
If index is `2` and total is `3`, it should return 3 items, currently it uses `[2:3]`, and returns 1 item. Let's fix that.

```
$ go run ./cmd/envtool tests shard --index=2 --total=3
^(TestAggregateCompatCollStats)
```
Also, if `total` or `index` is larger than the `testNames` slice, it panics. We should check that.